test/unit/org/apache/cassandra/tools/CMSOfflineToolTest.java

```diff
@@ -0,0 +1,688 @@
+        assertThat(outMetadata.tokenMap.tokens(joinedNodeId)).isEmpty();
+
+        assertCorrectEnvPostTest();
+    }
+
+    @Test
+    public void testForgetNodeInvalidNodeId() throws IOException
+    {
+        ClusterMetadata metadata = getThreeNodeClusterMetadata();
+        String metadataFile = dumpMetadata(metadata);
+        String outputFile = temporaryFolder.getRoot() + "/metadata-out.dump";
+        String nodeId = String.valueOf(-1);
+
+        ToolRunner.ToolResult result = ToolRunner.invokeClass(CMSOfflineTool.class,
+                                                              "forgetnode",
+                                                              "-f",
+                                                              metadataFile,
+                                                              "-id",
+                                                              nodeId,
+                                                              "-o",
+                                                              outputFile);
+
+
+        assertThat(result.getExitCode()).withFailMessage(result.getStderr()).isEqualTo(2);
+        assertThat(Files.exists(Paths.get(outputFile))).isFalse();
+        assertThat(result.getStderr()).contains("Node with id " + nodeId + " does not exist.");
+
+        assertCorrectEnvPostTest();
+    }
+
+    @Test
+    public void testPrintDataPlacements() throws IOException
+    {
+        ClusterMetadata metadata = getThreeNodeClusterMetadata();
+        String keyspaceName = "ks1";
+        KeyspaceParams ksParams = KeyspaceParams.create(true,
+                                                        Map.of("class", "NetworkTopologyStrategy",
+                                                               "datacenter1", "3"));
+        Keyspaces keyspaces = Keyspaces.none().with(KeyspaceMetadata.create(keyspaceName, ksParams));
+        DistributedSchema newSchema = new DistributedSchema(keyspaces);
+        metadata = updateMetadata(metadata, newSchema);
+
+        InetAddressAndPort inetAddressAndPort = InetAddressAndPort.getByNameUnchecked("127.0.0.1:7000");
+        InetAddressAndPort i2 = InetAddressAndPort.getByNameUnchecked("127.0.0.2:7000");
+        InetAddressAndPort i3 = InetAddressAndPort.getByNameUnchecked("127.0.0.3:7000");
+
+        DatabaseDescriptor.toolInitialization();
+        DatabaseDescriptor.setPartitionerUnsafe(metadata.partitioner);
+
+        DataPlacement dataPlacement = DataPlacement.builder()
+                                                   .withWriteReplica(Epoch.FIRST, Replica.fullReplica(inetAddressAndPort, new Range<>(metadata.partitioner.getMinimumToken(), metadata.partitioner.getTokenFactory().fromString("0"))))
+                                                   .withWriteReplica(Epoch.FIRST, Replica.fullReplica(i2, new Range<>(metadata.partitioner.getMinimumToken(), metadata.partitioner.getTokenFactory().fromString("0"))))
```

Comment:
nit: new Range<>(metadata.partitioner.getMinimumToken(), metadata.partitioner.getTokenFactory().fromString("0") can just be created once